cmd/xl-storage.go

```diff
@@ -2205,7 +2225,7 @@ func (s *xlStorage) writeAll(ctx context.Context, volume string, path string, b
 		return err
 	}
 
-	flags := os.O_CREATE | os.O_WRONLY | os.O_TRUNC
+	flags := os.O_CREATE | os.O_WRONLY
```

Comment:
I don't understand what the difference is by doing this by default. It seems like you are just adding another syscall to every write, and I don't see what the benefits are. 

If the file cannot be opened. Nothing happens. If the file is opened, it is truncated either way.

It seems to me you should

A) Keep `O_TRUNC`
B) Truncate(0) if the write fails (but *only* if it fails).

AFAICT that would do the same, but there wouldn't be any need for an additional syscall on the "happy path".

Does that make sense?

Reply:
That's correct, but it allows protection for situations where we don't make `xl.meta` 0 aggressively but safely later. 

Otherwise to do it another way is to do how we do `.minio.sys/tmp` and then rename()

Reply:
Let me revert this change, as its not related to this PR at all @klauspost